### ISSUE TITLE
Spring StopWatch show wrong classname.methodname in case of class hierarchy

### ISSUE THREAD

**Issue by lincetto** [virgo47/javasimon#14]
Hi, I've a class hierarchy like this (semplified)

```
interface Interface {
  @Monitored
  public run() {...}
}

class Class1 implements Interface {
  public run() {...}
}

class Class2 implements Interface {
  public run() {...}
}
```

The stopwatch uses as name the first class of which run method was invoked and then all invocations of run methods in both classes are traced with the same name and this is wrong.

Example : 

2015-07-29 12:08:34,976 DEBUG [Stopwatch] Split 1.03 s in Stopwatch Simon Stopwatch: total 879 s, counter 146, max 30.1 s, min 113 ms, mean 6.02 s [**Class1.run INHERIT "Class1.run"**]
because after the startup of the application I called Class1 run method first.

If I restart the application and call run method of Class2 before Class1, the Stopwatch will log :

2015-07-29 12:08:34,976 DEBUG [Stopwatch] Split 1.03 s in Stopwatch Simon Stopwatch: total 879 s, counter 146, max 30.1 s, min 113 ms, mean 6.02 s [**Class2.run INHERIT "Class2.run"**]

and all stopwatch for both run methods will be added together.

Obviously I can move the annotation to Class1 and Class2 run methods, but is it possible to solve this problem in order to avoid to write the annotation in every implementation of the interface?
Perhaps the problem is in how the CachedStopWatchSource use the location (interface method instead of target class method)?

Or, at least, the stopwatch should use the interface name + method name in order to not show misleading counters.

**Comment by virgo47** [comment]
I definitely agree this is a bug. I'll try to write test to reproduce it and then I'll fix it.

**Comment by virgo47** [comment]
Please, what version of Simon are you using?

**Comment by virgo47** [comment]
I added test where the situation is recreated, I believe. Please check https://github.com/virgo47/javasimon/blob/e6c4efb0a816957356652bf3a6e580ef64dddabe/spring/src/test/java/org/javasimon/spring/MonitoredPointcutStopwatchNameTest.java

It is based on Java Config, but that should not pose the problem, it uses the same classes, uses automatic discovery of `@Monitored` classes, adds the aspect to them, etc. But I cannot reproduce the problem itself, all the asserts work fine.

**Comment by lincetto** [comment]
Thank you very much for your helpfulness! Great job with JavaSimon!! :+1:

**Comment by virgo47** [comment]
Released today in 4.1.1
